#[Jump to my work in this notebook](#my_work)

In [1]:
import os
import sys
import numpy as np
#import pylab as pl


#import json
#import os
#s = json.load( open(os.getenv("PUI2015")+"/fbb_matplotlibrc.json") )
#pl.rcParams.update(s)

#%pylab inline

#from IPython.display import Image
#Image(filename='../plotsforclasses/NYCReentryprogram_title.png')



Image(filename='../plotsforclasses/NYCReentryprogram.png')

#NULL HYPOTHESIS: the % of former prisoners employed after release is the same or lower for candidates who participated in the program as for the control group, significance level p=0.05

#$H_0: P_0 - P_1 \geq$    0
    
#$H_a: P_0 - P_1 $< 0    
    
    
#$\alpha$ = 0.05    

##this is a TEST OF PROPORTIONS. we use the Binomial distribution since it is a yes/no (bernulli) test for each subject: the former inmate was or was not ever employed in a CEO transitional job (second row in the table above):
#$P_0=0.035, P_1=0.701$

In [3]:

alpha=0.05
#we like fractions better then percentages. as a rule of thumb, either use fractions or counts
P_0=3.5*0.01 
P_1=70.1*0.01

n_0=409
n_1=564

#lets get the counts by multiplying by the sample size
Nt_0=P_0*n_0
Nt_1=P_1*n_1


#2 samples, categorical data
#TWO OPTIONS z test, or chi square test.  

#START WITH Z TEST

##the z test compares the stanrard deviation of the expected distribution and the observed result. it tells you literally how many standard deviations from the tail an observation is, under the _assumption of normality
must define the sample standard deviation 

In [4]:
#define the sample proportion first
sp=(P_0*n_0+P_1*n_1)/(n_1+n_0)
print sp

0.421047276465


#standard deviation of the sampling distribution the distribution is Binomial, the binomail stdev is 

(see a proof here!: http://stats.stackexchange.com/questions/29641/standard-error-for-the-mean-of-a-sample-of-binomial-random-variables!): 

$\sqrt{\frac{p(1 - p)}{n}}$

for 2 samples this becomes 

$\sqrt{ \frac{ \hat{p}(1 - \hat{p})} {n1} + \frac{ \hat{p}(1 - \hat{p})} {n1} }$

cfr: page 138 of Statistics in a Nutshell, eq. 5.12 and here http://stattrek.com/hypothesis-test/difference-in-proportions.aspx?Tutorial=AP



Image("SiNS5.13.png")

Note that in the Image above, taken from the online version of the book, $\bar{x}$ should be $\hat{p}$!!

In [6]:
# i am goonna create a little one line function to calculate the standard dev, it is not really needed, but just to show you how you do such a thing
sp_stdev= lambda p, n: np.sqrt( p * ( 1 - p ) /n[0] +  p * ( 1 - p )/n[1]  )


sp_stdev_2y=sp_stdev((P_0+P_1),[n_0,n_1])
print P_0, n_0, n_1, sp_stdev_2y

0.035 409 564 0.028628334617


#z score: how many standard deviation away from the population parameter is my statistic?

#$z=\frac{P_1-P_0}{\sigma}$

In [7]:
zscore = lambda p0, p1, s : (p0-p1)/s
z_2y = zscore(P_1, P_0, sp_stdev_2y)
print z_2y


23.2636654877


Image('http://intersci.ss.uci.edu/wiki/images/3/3a/Normal01.jpg')

##if $p<\alpha$ : reject H0

##IMPORTANT!! note that this P is not the p-value, but 
##p-value=1-P

In [9]:
##p-value for employment after 2 years
p_2y=1-0.9998

def report_result(p,a):
    print 'is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p,a)
    if p<a:
        print "YES!"
    else: print "NO!"
    
    print 'the Null hypothesis is {}'.format( 'rejected' if p<a  else 'not rejected') 

    
report_result(p_2y,alpha)

is the p value 0.00 smaller than the critical value 0.05? 
YES!
the Null hypothesis is rejected


#!!!!!!!!TODO FOR YOU


<a id='my_work'></a>
#My work begins below.

##what if we used the values for where the former inmate was or was not ever employed in a CEO transitional job 3 years after release (5th row of the table)? 

#$P_0 = 27.2, P_1= 28.2$

#look up data table and insert the appropriate values to get the appropriate result!

In [2]:
#For Z test
alpha = 0.05
#we like fractions better then percentages. as a rule of thumb, either use fractions or counts
pCtrl = 27.2 * 0.01
pProg = 28.2 * 0.01

nCtrl = 409
nProg = 564

#Getting the counts by multiplying by the sample size
NtCtrl = pCtrl * nCtrl
NtProg = pProg * nProg
print NtCtrl, NtProg

111.248 159.048


In [3]:
#Defining sample proportion first
sampleProp = (pCtrl * nCtrl + pProg * nProg) / (nCtrl + nProg)
print sampleProp

0.277796505653


In [4]:
#For std deviation
sp_stdev = lambda p, n: np.sqrt(p * (1-p)/ n[0] + p * (1-p)/n[1])
sp_stdev_3y = sp_stdev((pCtrl + pProg), [nCtrl, nProg])
print pCtrl, nCtrl, nProg, sp_stdev_3y

0.272 409 564 0.0322832705591


In [5]:
#For Z-score (# of std deviations away from population parameter)
alpha = .05
zscore = lambda p0, p1, s : (p0-p1)/s
z_3y = zscore(pProg, pCtrl, sp_stdev_3y)
print z_3y

0.309757959054


In [6]:
#P-value for employment after 3 years

p_3y = 1 - 0.6368 #(z is close to .3 with alpha of .05)

def report_result(p, a):
    print 'Is the p value {0:.2f} smaller than the critical value {1:.2f}? '.format(p, a)
    if p < a:
        print "YES!" 
    else: print "NO!"
    
    print 'The null hypothesis is {}'.format('rejected.' if p < a else 'not rejected.') 

report_result(p_3y, alpha)

is the p value 0.36 smaller than the critical value 0.05? 
NO!
The null hypothesis is not rejected.


#Now lets do it with the $\chi^2$ test

#this can also be done with the $\chi^2$ distribution, see "Statistics In a Nutshell Chapter 4", or http://math.hws.edu/javamath/ryan/ChiSquare.html (if you are really just interested in the formula at face value)

##the chisq statistics tests a number against the distribution of the following quantity:

$$\chi^2 = \Sigma \frac{(observation - expectation)^2}{expectation}$$


if we talk about sample fractions  that is 

$$\chi^2 = \Sigma \frac{(f_{observed} - f_{expectated})^2}{f_{expected}}$$

##turns out this quantity is distributed according to a chi square distribution, so if i get the $\chi^2$ statistics i can compare it to the full chisq distribution and see how far in the tail it is

##the trickiest part (not that tricky) is to figure out the table of values. please see Statistics In a Nutshell Chapter 4, for our data for example:

|employed in subsadized job |     employed   | not employed   |                   
|---------------------------|----------------|------------------|---------------------------|
| test sample               | $70.1*5.64$    | $29.9*5.64$      | 564                       |
| control sample            | $3.5*4.09$     | $96.5*4.09$      | 409                       |
|                           |                |                  |                           |
| total                     | 409.679        |  562.912         | 973         |

the expected ratio is the product of the total of all rows and all columns, devided by the total

expected = $\frac{row tot * col tot}{total}$

In [7]:
Ntot = 973
expected = 564*409*409.679*562.912
sample_values = [[70.1*5.64,29.9*5.64],[3.05*4.09,96.5*4.09]]
 
chisqstat= lambda N, values, expect : N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expected)

print chisqstat(Ntot,  sample_values, expected)


433.44352059


#!!!!!! TODO FOR YOU!
#let's see what the chi sq statistics says about the employment for > 6 consecutive quarters (row 6)

|employed in subsidized job |    employed    |  not employed    |                   
|---------------------------|----------------|------------------|---------------------------|
| test sample               | $14.7*5.64$    | $85.3*5.64$      | 564                       |
| control sample            | $11.9*4.09$    | $88.1*4.09$      | 409                       |
|                           |                |                  |                           |
| total                     | 131.579        | 841.421          | 973                       |

In [8]:
#For Chi Square statistics
Ntot1 = 973
expected1 = 564*409*131.579*841.421
sample_values1 = [[14.7*5.64, 85.3*5.64], [11.9*4.09, 88.1*4.09]]

chisqstat1= lambda N, values, expect : N*((values[0][0]*values[1][1]-values[0][1]*values[1][0])**2)/(expected1)

print chisqstat1(Ntot1, sample_values1, expected1)

1.58939333193


Image("http://passel.unl.edu/Image/Namuth-CovertDeana956176274/chi-sqaure%20distribution%20table.PNG")

432 is hella larger than 3.84.
**Yes it is.**

Why am i mentioning 3.84?
**Because it is associated with the .05 confidence level with one degree of freedom.**

How does the six-quarter chi square statistic that you derived compare?
**The six-quarter chi square statistic is 1.589, which is less than 3.84.** 

Please state what that means in terms of your hypothesis in a markdown cell below!

##The null hypothesis cannot be rejected because the six-quarter chi square statistic is 1.589, which is less than 3.84.